### PR TITLE
Add json logger

### PR DESCRIPTION
**Pull request by ckaipf**
PR adds `JsonFormatter ` from `python-json-logger` to enable log serialization.

### CODE REVIEW

File: datameta/siteid.py [hunk 1]
```diff
@@ -28,13 +28,13 @@
     try:
         _prefix[entity] = threadlocal.get_current_registry().settings[f'datameta.site_id_prefix.{entity}']
     except Exception as e:
```
Comment (ckaipf) [inline datameta/siteid.py:30]: Please review, not sure everything is intentionally here

File: datameta/api/ui/forgot.py [hunk 2]
```diff
@@ -40,7 +40,7 @@ def send_forgot_token(request, db_token_obj, clear_token):
         )
 
     if asbool(threadlocal.get_current_registry().settings['datameta.logging.log_token_urls']):
```
[the remark this answers is not in the document]
Comment (lkuchenb) [reply datameta/api/ui/forgot.py:42]: As discussed, feel free to remove the check

File: datameta/api/ui/forgot.py [hunk 3]
```diff
@@ -40,7 +40,7 @@ def send_forgot_token(request, db_token_obj, clear_token):
...
-        log.debug(f"USER REQUESTED RECOVERY TOKEN: '{token_url}'")
+        log.debug("User requsted recovery token.", extra={"token_url": token_url})
```
[the remark this answers is not in the document]
Comment (lkuchenb) [reply datameta/api/ui/forgot.py:43]: Let's keep this the same style as with the forgot_api call

File: datameta/email/__init__.py [hunk 4]
```diff
@@ -49,4 +49,4 @@ def send(recipients, subject, template, values, bcc=None, rec_header_only=False)
     try:
         __smtp.sendMessage(__smtp_from, recipients, subject, message, bcc=bcc, rec_header_only=rec_header_only)
     except Exception as e:
-        log.error(f"An error occurred when sending an email: {e}")
+        log.error("Email sending exception.", extra={"exception": e})
```
Comment (lkuchenb) [inline datameta/email/__init__.py:52]:
```suggestion
        log.error("SMTP error.", extra={"error": e})
```

File: datameta/siteid.py [hunk 5]
```diff
@@ -44,6 +44,6 @@ def generate(request, BaseClass):
     for _ in range(10):
         new_id = prefix + str(random.randint(0, pow(10, digits))).rjust(digits, "0")
         if request.dbsession.query(BaseClass).filter(BaseClass.site_id == new_id).first():
-            log.warning(f"Site ID collision for {BaseClass.__tablename__}. Your ID space may be saturating.")
+            log.warning("Site ID collision, ID space may saturates.", extra={"tablename": BaseClass.__tablename__})
```
Comment (lkuchenb) [inline datameta/siteid.py:47]:
```suggestion
            log.warning("Site ID collision, ID space may be saturating.", extra={"tablename": BaseClass.__tablename__})
```

File: datameta/views/login.py [hunk 6]
```diff
@@ -35,7 +35,7 @@ def my_view(request):
 
             auth_user = security.get_user_by_credentials(request, in_email, in_pwd)
             if auth_user:
-                log.info(f"LOGIN [uid={auth_user.id},email={auth_user.email}] FROM [{request.client_addr}]")
+                log.info("Login.", extra={"user_id": auth_user.id, "email": auth_user.email, "client_addr": request.client_addr})
```
Comment (lkuchenb) [inline datameta/views/login.py:38]:
```suggestion
                log.info("User logged in.", extra={"user_id": auth_user.id, "email": auth_user.email, "client_addr": request.client_addr})
```

**Review by lkuchenb — CHANGES_REQUESTED** [review]
Thanks, see remarks and suggestions above!

**Review by lkuchenb — APPROVED** [review]

**Pull request merged**